proxy/grpcproxy/cluster.go

```diff
@@ -15,20 +15,99 @@
 package grpcproxy
 
 import (
+	"fmt"
+	"sync"
+
 	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/naming"
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 
 	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	gnaming "google.golang.org/grpc/naming"
 )
 
 type clusterProxy struct {
 	client *clientv3.Client
```

Comment:
> How can we not use clientv3.Client when we need to use session via clientv3.Client in register?
`register` doesn't directly use `clusterProxy`, it tracks `clientv3.Client` on its own.

>  it would need additional type case
An extra typecast is OK. Nothing should be hammering the cluster API anyway.